### PR TITLE
fix delete for arrays and minor fixes

### PR DESCRIPTION
**Pull request by apbard**
deleting arrays requires delete[]. 
- some minor fixes: missing consts and  &

### CODE REVIEW

File: Mediator.h [hunk 1]
```diff
@@ -10,13 +10,12 @@ template <typename Item> class Mediator
 {
 public:
   
-  Mediator( int nItems )
+  Mediator(int nItems ):N(nItems)
```
Comment (craffel) [inline Mediator.h:13]: What does the `:N(nItems)` do? I haven't seen that syntax before (granted, my C++ is very rusty...)

Comment (apbard) [inline Mediator.h:13]: It is an initialization list. when you call a constructor it takes care of the instantiation of all class member before executing the code itself. Using an initialization list allow to specify how to instatiate those members by passing arguments to their constructors. This avoids instatiation with defaults just for reallocating/change them in constructor body.
In this case, for instance the original code is equivalent to: int N; N = nItems;   the new instead to int N = nItems;
The overhead is negligible for ints but maybe huge for some custom class. 
Also the initialization list is where you call constructors of an hypotetical base class.

if you would like to have more details see for instance:
http://www.developerfusion.com/article/133063/constructors-in-c11/

Comment (craffel) [inline Mediator.h:13]: Cool, got it, thanks for the explanation!

**Comment by craffel** [comment]
Thanks for this, merging!

**Pull request merged**